discovery-service/src/main/java/org/zowe/apiml/discovery/GatewayNotifier.java

```diff
@@ -14,30 +14,66 @@
 import com.netflix.eureka.EurekaServerContext;
 import com.netflix.eureka.EurekaServerContextHolder;
 import com.netflix.eureka.registry.PeerAwareInstanceRegistry;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.util.BiConsumer;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
 import org.springframework.stereotype.Component;
 import org.springframework.web.client.RestTemplate;
 import org.zowe.apiml.message.core.MessageService;
 import org.zowe.apiml.message.log.ApimlLogger;
 import org.zowe.apiml.util.EurekaUtils;
 
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.Collections;
 import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
 
+/**
+ * This bean is used to send a notification to the Gateways. It is sending notification asynchronous (at first they
+ * are stored into the queue and then with a delay send to Gateways). If same notification is waiting for sending
+ * it send it only once.
+ *
+ * Purpose of this bean is at first in notification Gateways about new and removed services and process at least
+ * evicting of caches there.
+ */
 @Component
 @Slf4j
-public class GatewayNotifier {
+public class GatewayNotifier implements Runnable {
+
+    public static final String GATEWAY_SERVICE_ID = "GATEWAY";
```

Comment:
You can use Gateway constant from `org.zowe.apiml.product.constants.CoreService` instead.

Reply:
Fixed, just here is needed uppercase value, so I just replace the literal, constant itself is still there.